lib/signs_ui/alerts/state.ex

```diff
@@ -24,6 +24,20 @@ defmodule SignsUi.Alerts.State do
     {:ok, %__MODULE__{}}
   end
 
+  @spec active_alert_ids(GenServer.server()) :: [Alert.id()]
+  def active_alert_ids(pid \\ __MODULE__) do
+    GenServer.call(pid, :active_alert_ids)
+  end
+
+  def handle_call(:active_alert_ids, _from, state) do
+    alerts = state.alerts
+    routes = Map.keys(alerts)
+    alert_ids = for route <- routes, do: Map.keys(alerts[route])
```

Comment:
`for` expressions are quite powerful. You can deconstruct further by adding more expressions in more commas, and you can deconstruct a map into key-value pairs. So this, should get you a nice flat list of alert_ids in one go:

```ex
alert_ids =
  for {route_id, route_alerts} <- state.alerts,
      {alert_id, _alert_data} <- route_alerts,
      do: alert_id
```